packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -105,19 +105,29 @@
     }
   }
 
-  $: fetchRows(searchTerm, primaryDisplay)
+  $: fetchRows(searchTerm, primaryDisplay, defaultValue)
 
-  const fetchRows = (searchTerm, primaryDisplay) => {
+  const fetchRows = async (searchTerm, primaryDisplay, defaultVal) => {
     const allRowsFetched =
       $fetch.loaded &&
       !Object.keys($fetch.query?.string || {}).length &&
       !$fetch.hasNextPage
-    // Don't request until we have the primary display
-    if (!allRowsFetched && primaryDisplay) {
-      fetch.update({
-        query: { string: { [primaryDisplay]: searchTerm } },
+    // Don't request until we have the primary display or default value has been fetched
+    if (allRowsFetched || !primaryDisplay) {
+      return
+    }
+    if (defaultVal) {
```

Comment:
I think we need to check here if we already have the row we need for the default value. Otherwise we're going to do a search for the default value row every time, which blocks the main search call and means we're spending twice as long waiting.

Reply:
Yeap - Adri raised this - fixed now!